pinecone/integration_test_suite.go

```diff
@@ -0,0 +1,218 @@
+package pinecone
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type IntegrationTests struct {
+	suite.Suite
+	apiKey            string
+	client            *Client
+	host              string
+	dimension         int32
+	indexType         string
+	vectorIds         []string
+	podIdxName        string
+	serverlessIdxName string
```

Comment:
What's the reason for needing both of these names in each struct? It seems like in `run_integration_test_suites.go`, we're creating two different `IntegrationTests` objects. Do they not run separately?

I'm just thinking it feels easier to reason about each instance of the struct handling it's own index. I know you mentioned both serverless and pod index tests run regardless of what file you're trying to test, so there might be something I'm misunderstanding. Basically, my initial thoughts were "why can't this just be `idxName` and each suite handles one index?

Reply:
Yeah....I think you're right! Lemme try it out and see if CI still passes...